// @fivethree/core/esm2015/lib/gallery/gallery-toolbar/gallery-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/gallery/gallery-toolbar/gallery-toolbar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, TemplateRef, ViewChild, Input } from '@angular/core';
export class FivGalleryToolbar {
    constructor() {
        this.position = 'top';
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
}
FivGalleryToolbar.decorators = [
    { type: Component, args: [{
                selector: 'fiv-gallery-toolbar',
                template: "<ng-template #ref>\n  <ng-content></ng-content>\n</ng-template>",
                styles: [""]
            }] }
];
/** @nocollapse */
FivGalleryToolbar.ctorParameters = () => [];
FivGalleryToolbar.propDecorators = {
    content: [{ type: ViewChild, args: [TemplateRef, { static: false },] }],
    position: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FivGalleryToolbar.prototype.content;
    /** @type {?} */
    FivGalleryToolbar.prototype.position;
}
