// @fivethree/core/esm5/lib/app-bar/app-bar-tab/app-bar-tab.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/app-bar/app-bar-tab/app-bar-tab.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Host, HostBinding, ElementRef } from '@angular/core';
import { Router } from '@angular/router';
import { FivAppBar } from '../app-bar.component';
var FivAppBarTab = /** @class */ (function () {
    function FivAppBarTab(appBar, router, el) {
        this.appBar = appBar;
        this.router = router;
        this.el = el;
        this.type = 'unbounded';
        this.badge = -1;
    }
    Object.defineProperty(FivAppBarTab.prototype, "active", {
        get: /**
         * @return {?}
         */
        function () {
            return this.router.url.endsWith(this.href);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FivAppBarTab.prototype, "classes", {
        get: /**
         * @return {?}
         */
        function () {
            return "label-" + this.appBar.titleLayout;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FivAppBarTab.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    FivAppBarTab.prototype.onClick = /**
     * @return {?}
     */
    function () {
        this.appBar.ionTabButtonClick({
            tab: this.tab,
            href: this.href,
            selected: true
        });
    };
    FivAppBarTab.decorators = [
        { type: Component, args: [{
                    selector: 'fiv-app-bar-tab',
                    template: "<ng-template>\n  <ng-content></ng-content>\n</ng-template>\n\n<div (click)=\"onClick()\" class=\"tab ion-activatable\" [ngClass]=\"{'active': active}\">\n  <ng-content select=\"fiv-icon\"></ng-content>\n  <ng-content *ngIf=\"appBar.titleLayout === 'show' || (active && appBar.titleLayout === 'active-only')\"\n    select=\"ion-label\"></ng-content>\n  <ion-ripple-effect [type]=\"type\"></ion-ripple-effect>\n</div>",
                    styles: [":host{width:50%;height:100%;display:inline-block;position:relative}:host .tab{width:100%;height:100%;display:flex;position:relative;flex-direction:column;align-items:center;justify-content:center;overflow:visible}:host .tab.active{--fiv-color-icon:var(--ion-color-primary)}:host .tab.active ::ng-deep ion-label{color:var(--ion-color-primary)}:host(.label-hide) ::ng-deep ion-label{display:none}:host(.label-active-only) .tab:not(.active) ::ng-deep ion-label{display:none}"]
                }] }
    ];
    /** @nocollapse */
    FivAppBarTab.ctorParameters = function () { return [
        { type: FivAppBar, decorators: [{ type: Host }] },
        { type: Router },
        { type: ElementRef }
    ]; };
    FivAppBarTab.propDecorators = {
        name: [{ type: Input }],
        tab: [{ type: Input }],
        href: [{ type: Input }],
        type: [{ type: Input }],
        badge: [{ type: Input }],
        classes: [{ type: HostBinding, args: ['class',] }]
    };
    return FivAppBarTab;
}());
export { FivAppBarTab };
if (false) {
    /** @type {?} */
    FivAppBarTab.prototype.name;
    /** @type {?} */
    FivAppBarTab.prototype.tab;
    /** @type {?} */
    FivAppBarTab.prototype.href;
    /** @type {?} */
    FivAppBarTab.prototype.type;
    /** @type {?} */
    FivAppBarTab.prototype.badge;
    /** @type {?} */
    FivAppBarTab.prototype.appBar;
    /** @type {?} */
    FivAppBarTab.prototype.router;
    /** @type {?} */
    FivAppBarTab.prototype.el;
}
