// @fivethree/core/esm5/lib/overlay/overlay-content/overlay-content.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/overlay/overlay-content/overlay-content.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, HostBinding } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
var FivOverlayContent = /** @class */ (function () {
    function FivOverlayContent(sanitizer) {
        this.sanitizer = sanitizer;
    }
    Object.defineProperty(FivOverlayContent.prototype, "myStyle", {
        get: /**
         * @return {?}
         */
        function () {
            return this.priority
                ? this.sanitizer.bypassSecurityTrustStyle("z-index: " + this.priority)
                : null;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    FivOverlayContent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    FivOverlayContent.decorators = [
        { type: Component, args: [{
                    selector: 'fiv-overlay-content',
                    template: "<ng-content #content></ng-content>",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    FivOverlayContent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    FivOverlayContent.propDecorators = {
        myStyle: [{ type: HostBinding, args: ['style',] }]
    };
    return FivOverlayContent;
}());
export { FivOverlayContent };
if (false) {
    /** @type {?} */
    FivOverlayContent.prototype.priority;
    /**
     * @type {?}
     * @private
     */
    FivOverlayContent.prototype.sanitizer;
}
