// @fivethree/core/esm5/lib/gallery/gallery-toolbar/gallery-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/gallery/gallery-toolbar/gallery-toolbar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, TemplateRef, ViewChild, Input } from '@angular/core';
var FivGalleryToolbar = /** @class */ (function () {
    function FivGalleryToolbar() {
        this.position = 'top';
    }
    /**
     * @return {?}
     */
    FivGalleryToolbar.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    FivGalleryToolbar.decorators = [
        { type: Component, args: [{
                    selector: 'fiv-gallery-toolbar',
                    template: "<ng-template #ref>\n  <ng-content></ng-content>\n</ng-template>",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    FivGalleryToolbar.ctorParameters = function () { return []; };
    FivGalleryToolbar.propDecorators = {
        content: [{ type: ViewChild, args: [TemplateRef, { static: false },] }],
        position: [{ type: Input }]
    };
    return FivGalleryToolbar;
}());
export { FivGalleryToolbar };
if (false) {
    /** @type {?} */
    FivGalleryToolbar.prototype.content;
    /** @type {?} */
    FivGalleryToolbar.prototype.position;
}
