// @fivethree/core/esm5/lib/popover/popover-element.directive.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/popover/popover-element.directive.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { FivPopover } from './popover.component';
import { Directive, ElementRef, Input } from '@angular/core';
import { tap, delay } from 'rxjs/operators';
var PopoverElementDirective = /** @class */ (function () {
    function PopoverElementDirective(el) {
        this.el = el;
        this.delay = 50;
    }
    Object.defineProperty(PopoverElementDirective.prototype, "show", {
        get: /**
         * @return {?}
         */
        function () {
            return this._show;
        },
        set: /**
         * @param {?} visible
         * @return {?}
         */
        function (visible) {
            var _this = this;
            this._show = visible;
            setTimeout((/**
             * @return {?}
             */
            function () {
                if (_this.show) {
                    _this.fivPopover.open(_this.el);
                }
                else if (_this.fivPopover.overlay && _this.fivPopover.overlay.open) {
                    _this.fivPopover.close();
                }
            }), 0);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    PopoverElementDirective.prototype.open = /**
     * @return {?}
     */
    function () {
        this.show = true;
    };
    /**
     * @return {?}
     */
    PopoverElementDirective.prototype.close = /**
     * @return {?}
     */
    function () {
        this.show = false;
    };
    /**
     * @return {?}
     */
    PopoverElementDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.fivPopover.overlay.afterInit
            .pipe(delay(this.delay), tap((/**
         * @return {?}
         */
        function () {
            _this.show = _this._show;
        })))
            .subscribe();
    };
    PopoverElementDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[fivPopover]',
                    exportAs: 'fivPopover'
                },] }
    ];
    /** @nocollapse */
    PopoverElementDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    PopoverElementDirective.propDecorators = {
        delay: [{ type: Input, args: ['fivPopover.delay',] }],
        fivPopover: [{ type: Input }],
        show: [{ type: Input, args: ['fivPopover.show',] }]
    };
    return PopoverElementDirective;
}());
export { PopoverElementDirective };
if (false) {
    /** @type {?} */
    PopoverElementDirective.prototype._show;
    /** @type {?} */
    PopoverElementDirective.prototype.delay;
    /** @type {?} */
    PopoverElementDirective.prototype.fivPopover;
    /**
     * @type {?}
     * @private
     */
    PopoverElementDirective.prototype.el;
}
