// @fivethree/core/esm5/lib/overflow-buttons/overflow-buttons.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/overflow-buttons/overflow-buttons.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { FivPopover } from './../popover/popover.component';
import { Platform, PopoverController, ActionSheetController } from '@ionic/angular';
import { Component, Input, ContentChildren, QueryList } from '@angular/core';
import { FivButton } from '../button/button.component';
var FivOverflowButtons = /** @class */ (function () {
    function FivOverflowButtons(platform, popoverController, actionSheetController) {
        this.platform = platform;
        this.popoverController = popoverController;
        this.actionSheetController = actionSheetController;
        this.slot = 'end';
        this.icon = 'more';
        this.mode = 'popover';
        this.count = 0;
    }
    /**
     * @return {?}
     */
    FivOverflowButtons.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    FivOverflowButtons.prototype.ngAfterContentInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @param {?} event
     * @return {?}
     */
    FivOverflowButtons.prototype.onMenuClicked = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.mode === 'action-sheet') {
            return this.presentActionSheet(event);
        }
        if (this.mode === 'popover') {
            return this.presentPopover(event);
        }
        if (this.platform.is('mobile')) {
            this.presentActionSheet(event);
        }
        else {
            this.presentPopover(event);
        }
    };
    /**
     * @param {?} ev
     * @return {?}
     */
    FivOverflowButtons.prototype.presentPopover = /**
     * @param {?} ev
     * @return {?}
     */
    function (ev) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var popover, result;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.popoverController.create({
                            component: FivPopover,
                            event: ev,
                            translucent: false,
                            componentProps: {
                                buttons: this.buttons
                                    .toArray()
                                    .slice(this.count)
                                    .filter((/**
                                 * @param {?} button
                                 * @return {?}
                                 */
                                function (button) { return !button.disabled; }))
                            }
                        })];
                    case 1:
                        popover = _a.sent();
                        return [4 /*yield*/, popover.present()];
                    case 2:
                        _a.sent();
                        return [4 /*yield*/, popover.onDidDismiss()];
                    case 3:
                        result = _a.sent();
                        if (result &&
                            result.data &&
                            (result.data.index || result.data.index === 0)) {
                            this.onButtonClicked(result.data.index + this.count);
                        }
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} ev
     * @return {?}
     */
    FivOverflowButtons.prototype.presentActionSheet = /**
     * @param {?} ev
     * @return {?}
     */
    function (ev) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var buttons, actionSheet;
            var _this = this;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        buttons = [];
                        this.buttons
                            .toArray()
                            .slice(this.count)
                            .forEach((/**
                         * @param {?} button
                         * @param {?} index
                         * @return {?}
                         */
                        function (button, index) {
                            if (!button.disabled) {
                                buttons.push({
                                    text: button.text,
                                    icon: button.icon,
                                    handler: (/**
                                     * @return {?}
                                     */
                                    function () {
                                        _this.onButtonClicked(_this.count + index);
                                    })
                                });
                            }
                        }));
                        return [4 /*yield*/, this.actionSheetController.create({
                                buttons: buttons
                            })];
                    case 1:
                        actionSheet = _a.sent();
                        return [4 /*yield*/, actionSheet.present()];
                    case 2:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    /**
     * @param {?} i
     * @return {?}
     */
    FivOverflowButtons.prototype.onButtonClicked = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        this.buttons.toArray()[i].click.emit();
    };
    FivOverflowButtons.decorators = [
        { type: Component, args: [{
                    selector: 'fiv-overflow-buttons',
                    template: "<ion-buttons *ngIf=\"buttons\" [slot]=\"slot\">\n\n  <fiv-button [color]=\"color\" *ngIf=\"(slot === 'start' || slot ==='primary') && count < buttons.length && text\"\n    (click)=\"onMenuClicked($event)\" [text]=\"text\"></fiv-button>\n\n  <fiv-button [color]=\"color\" *ngIf=\"(slot === 'start' || slot ==='primary') && count < buttons.length && !text\"\n    (click)=\"onMenuClicked($event)\" [icon]=\"'md-more'\"></fiv-button>\n  <ng-container>\n    <fiv-button [color]=\"button.color\" (click)=\"onButtonClicked(i)\" [icon]=\"button.icon\"\n      *ngFor=\"let button of buttons.toArray() |\u00A0slice:0:count; let i = index;\"></fiv-button>\n  </ng-container>\n  <fiv-button [color]=\"color\" *ngIf=\"(slot === 'end' || slot ==='secondary') && count < buttons.length && !text\"\n    (click)=\"onMenuClicked($event)\" [icon]=\"icon\"></fiv-button>\n  <fiv-button [color]=\"color\" *ngIf=\"(slot === 'end' || slot ==='secondary') && count < buttons.length && text\"\n    (click)=\"onMenuClicked($event)\" [text]=\"text\"></fiv-button>\n</ion-buttons>",
                    styles: [":host{display:block}"]
                }] }
    ];
    /** @nocollapse */
    FivOverflowButtons.ctorParameters = function () { return [
        { type: Platform },
        { type: PopoverController },
        { type: ActionSheetController }
    ]; };
    FivOverflowButtons.propDecorators = {
        slot: [{ type: Input }],
        icon: [{ type: Input }],
        text: [{ type: Input }],
        mode: [{ type: Input }],
        color: [{ type: Input }],
        count: [{ type: Input }],
        buttons: [{ type: ContentChildren, args: [FivButton,] }]
    };
    return FivOverflowButtons;
}());
export { FivOverflowButtons };
if (false) {
    /** @type {?} */
    FivOverflowButtons.prototype.slot;
    /** @type {?} */
    FivOverflowButtons.prototype.icon;
    /** @type {?} */
    FivOverflowButtons.prototype.text;
    /** @type {?} */
    FivOverflowButtons.prototype.mode;
    /** @type {?} */
    FivOverflowButtons.prototype.color;
    /** @type {?} */
    FivOverflowButtons.prototype.count;
    /** @type {?} */
    FivOverflowButtons.prototype.buttons;
    /**
     * @type {?}
     * @private
     */
    FivOverflowButtons.prototype.platform;
    /**
     * @type {?}
     * @private
     */
    FivOverflowButtons.prototype.popoverController;
    /** @type {?} */
    FivOverflowButtons.prototype.actionSheetController;
}
