// @fivethree/core/esm2015/lib/button/button.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/button/button.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, EventEmitter, Output } from '@angular/core';
export class FivButton {
    constructor() {
        this.click = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
}
FivButton.decorators = [
    { type: Component, args: [{
                selector: 'fiv-button',
                template: "<ion-button shape=\"round\" [color]=\"color\" [disabled]=\"disabled\">\n  <fiv-icon [slot]=\"text ? 'start' : 'icon-only'\" [name]=\"icon\"></fiv-icon>\n  {{text}}\n</ion-button>",
                styles: [""]
            }] }
];
/** @nocollapse */
FivButton.ctorParameters = () => [];
FivButton.propDecorators = {
    icon: [{ type: Input }],
    text: [{ type: Input }],
    color: [{ type: Input }],
    disabled: [{ type: Input }],
    click: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    FivButton.prototype.icon;
    /** @type {?} */
    FivButton.prototype.text;
    /** @type {?} */
    FivButton.prototype.color;
    /** @type {?} */
    FivButton.prototype.disabled;
    /** @type {?} */
    FivButton.prototype.click;
}
