// @fivethree/core/esm5/lib/app-bar/app-bar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/app-bar/app-bar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { FivFab } from './../fab/fab.component';
import { Component, Input, ViewChild, Output, EventEmitter, Host, ContentChild } from '@angular/core';
import { Router } from '@angular/router';
import { IonTabs } from '@ionic/angular';
import { FivAppBarFabDirective } from './app-bar-fab.directive';
import { Subject } from 'rxjs';
var FivAppBar = /** @class */ (function () {
    function FivAppBar(router, ionTabs) {
        this.router = router;
        this.ionTabs = ionTabs;
        this._fabVisible = true;
        this.cutoutVisible = true;
        this.transitioning = false;
        this.titleLayout = 'hide';
        this.fivFabClick = new EventEmitter();
        this.onDestroy$ = new Subject();
    }
    /**
     * @return {?}
     */
    FivAppBar.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    FivAppBar.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.onDestroy$.next();
    };
    /**
     * @return {?}
     */
    FivAppBar.prototype.ngAfterContentInit = /**
     * @return {?}
     */
    function () {
        // merge(...this.tabs.map(tab => tab.ionTabButtonClick))
        //   .pipe(
        //     tap((event: TabButtonClickEventDetail) =>
        //       this.ionTabButtonClick(event)
        //     ),
        //     takeUntil(this.onDestroy$)
        //   )
        //   .subscribe();
    };
    /**
     * @return {?}
     */
    FivAppBar.prototype.fabClick = /**
     * @return {?}
     */
    function () {
        this.fivFabClick.emit(this);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    FivAppBar.prototype.ionTabButtonClick = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.ionTabs.select(event.tab);
    };
    FivAppBar.decorators = [
        { type: Component, args: [{
                    selector: 'fiv-app-bar',
                    template: "<div class=\"appbar\"\n  [ngClass]=\"{'cutout': cutoutVisible, 'left': fivFab.fivAppBarFab === 'left', 'right': fivFab.fivAppBarFab === 'right'}\">\n  <div class=\"appbar-rect left\">\n    <div class=\"tabs\">\n      <ng-content select=\"[left]\"></ng-content>\n    </div>\n  </div>\n  <ng-content select=\"fiv-fab\"></ng-content>\n  <svg viewBox=\"0 0 100 50\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n    <defs>\n      <path d=\"M0,0 C0,28 22,50 50,50 C78,50 100,28 100,0 L101,50 L0,50\" id=\"path-1\">\n      </path>\n    </defs>\n    <g id=\"Page-1\" stroke=\"none\" stroke-width=\"1px\" fill=\"none\" fill-rule=\"evenodd\">\n      <g id=\"cutout\">\n        <mask id=\"mask-2\" fill=\"white\">\n          <use xlink:href=\"#path-1\"></use>\n        </mask>\n        <g id=\"Path\"></g>\n        <polygon id=\"Path\" fill-rule=\"nonzero\" mask=\"url(#mask-2)\" points=\"0 0 100 0 100 50 0 50\">\n        </polygon>\n      </g>\n    </g>\n  </svg>\n  <div class=\"rect-filler\"></div>\n  <div class=\"appbar-rect right\">\n    <div class=\"tabs\">\n\n      <ng-content select=\"[right]\"></ng-content>\n    </div>\n  </div>\n</div>",
                    styles: [":host{display:block;position:relative;--fiv-radius:72px;--fiv-appbar-fab-size:56px;--fiv-appbar-height:56px;height:calc(var(--fiv-appbar-height) + env(safe-area-inset-bottom) + var(--fiv-appbar-fab-size)/ 2);width:100%;--fiv-border-radius:6px;--fiv-color-appbar:var(--ion-color-light);--fiv-side-width:12px}svg{height:calc(var(--fiv-radius)/ 2);width:var(--fiv-radius);left:calc(50% - (var(--fiv-radius)/ 2));position:absolute;top:calc(var(--fiv-appbar-fab-size)/ 2 + var(--fiv-border-radius))}svg polygon{fill:var(--fiv-color-appbar)}.appbar{width:100%;position:absolute;bottom:0;left:0;height:100%;background:rgba(0,0,0,0);z-index:5000}.appbar .appbar-rect{overflow:hidden;position:absolute;width:calc(50% - (var(--fiv-radius)/ 2));height:calc(var(--fiv-appbar-height) + env(safe-area-inset-bottom));bottom:0;box-shadow:3px -4px 5px 0 rgba(0,0,0,.1)}.appbar .appbar-rect.left{left:0;background:var(--fiv-color-appbar);border-top-right-radius:var(--fiv-border-radius)}.appbar .appbar-rect.right{right:0;background:var(--fiv-color-appbar);border-top-left-radius:var(--fiv-border-radius)}.appbar ::ng-deep fiv-fab{position:absolute;left:calc(50% - .5 * var(--fiv-fab-size));top:var(--fiv-border-radius);display:block;--fiv-fab-size:var(--fiv-appbar-fab-size);z-index:14}.appbar ::ng-deep fiv-fab fiv-icon{--fiv-color-icon:var(--fiv-color-fab-icon)}.appbar .rect-filler{width:var(--fiv-radius);height:calc(env(safe-area-inset-bottom) + var(--fiv-appbar-height) - var(--fiv-radius)/ 2 - var(--fiv-border-radius));position:absolute;overflow:hidden;bottom:0;left:calc(50% - (var(--fiv-radius)/ 2));background:var(--fiv-color-appbar)}.appbar:not(.cutout) .appbar-rect.left{border-top-right-radius:0}.appbar:not(.cutout) .appbar-rect.right{border-top-left-radius:0}.appbar.left .appbar-rect.left{width:var(--fiv-side-width)}.appbar.left .appbar-rect.right{width:calc(100% - var(--fiv-radius) - var(--fiv-side-width))}.appbar.left .rect-filler,.appbar.left svg{left:var(--fiv-side-width)}.appbar.left ::ng-deep fiv-fab{transform:translateX(0);left:calc(var(--fiv-side-width) + var(--fiv-radius)/ 2 - var(--fiv-appbar-fab-size)/ 2);position:absolute}.appbar.right .appbar-rect.left{width:calc(100% - var(--fiv-radius) - var(--fiv-side-width))}.appbar.right .appbar-rect.right{width:var(--fiv-side-width)}.appbar.right .rect-filler,.appbar.right svg{left:calc(100% - var(--fiv-radius) - var(--fiv-side-width))}.appbar.right ::ng-deep fiv-fab{transform:translateX(0);left:calc(100% - var(--fiv-side-width) - var(--fiv-radius)/ 2 - var(--fiv-appbar-fab-size)/ 2)}.tabs{display:flex;width:100%;height:100%;position:relative}"]
                }] }
    ];
    /** @nocollapse */
    FivAppBar.ctorParameters = function () { return [
        { type: Router },
        { type: IonTabs, decorators: [{ type: Host }] }
    ]; };
    FivAppBar.propDecorators = {
        fab: [{ type: ViewChild, args: ['fab', { static: false },] }],
        titleLayout: [{ type: Input }],
        fivFabClick: [{ type: Output }],
        fivFab: [{ type: ContentChild, args: [FivAppBarFabDirective, { static: false },] }]
    };
    return FivAppBar;
}());
export { FivAppBar };
if (false) {
    /** @type {?} */
    FivAppBar.prototype._fabVisible;
    /** @type {?} */
    FivAppBar.prototype.cutoutVisible;
    /** @type {?} */
    FivAppBar.prototype._position;
    /** @type {?} */
    FivAppBar.prototype.transitioning;
    /** @type {?} */
    FivAppBar.prototype.fab;
    /** @type {?} */
    FivAppBar.prototype.titleLayout;
    /** @type {?} */
    FivAppBar.prototype.fivFabClick;
    /** @type {?} */
    FivAppBar.prototype.fivFab;
    /** @type {?} */
    FivAppBar.prototype.onDestroy$;
    /** @type {?} */
    FivAppBar.prototype.router;
    /** @type {?} */
    FivAppBar.prototype.ionTabs;
}
