// @fivethree/core/esm2015/lib/expandable/expandable-indicator/expandable-indicator.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/expandable/expandable-indicator/expandable-indicator.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { FivExpandable } from '../expandable.component';
export class FivExpandableIndicator {
    /**
     * @param {?} expandable
     */
    constructor(expandable) {
        this.expandable = expandable;
        this.icon = 'ios-arrow-down';
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
}
FivExpandableIndicator.decorators = [
    { type: Component, args: [{
                selector: 'fiv-expandable-indicator',
                template: "<ion-icon class=\"indicator\" [ngClass]=\"{'rotate': expandable.isOpen}\" [name]=\"icon\"></ion-icon>",
                styles: [":host{display:flex}.indicator{transition:125ms}.indicator.rotate{transform:rotateZ(180deg)}"]
            }] }
];
/** @nocollapse */
FivExpandableIndicator.ctorParameters = () => [
    { type: FivExpandable }
];
FivExpandableIndicator.propDecorators = {
    icon: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FivExpandableIndicator.prototype.icon;
    /** @type {?} */
    FivExpandableIndicator.prototype.expandable;
}
